pyat/at/physics/nonlinear.py

```diff
@@ -193,12 +235,165 @@ def gen_detuning_elem(ring: Lattice, orbit: Optional[Orbit] = None) -> Element:
         orbit, _ = find_orbit(ring)
     lindata0, _, _ = ring.linopt6(get_chrom=False, orbit=orbit)
     xsi = get_chrom(ring.radiation_off(copy=True))
-    r0, r1, x, q_dx, y, q_dy = detuning(ring.radiation_off(copy=True),
-                                        xm=1.0e-4, ym=1.0e-4, npoints=3)
-    nonlin_elem = Element('NonLinear', PassMethod='DeltaQPass',
-                          Betax=lindata0.beta[0], Betay=lindata0.beta[1],
-                          Alphax=lindata0.alpha[0], Alphay=lindata0.alpha[1],
-                          Qpx=xsi[0], Qpy=xsi[1],
-                          A1=r1[0][0], A2=r1[0][1], A3=r1[1][1],
-                          T1=-orbit, T2=orbit)
+    r0, r1, x, q_dx, y, q_dy = detuning(
+        ring.radiation_off(copy=True), xm=1.0e-4, ym=1.0e-4, npoints=3
+    )
+    nonlin_elem = Element(
+        "NonLinear",
+        PassMethod="DeltaQPass",
+        Betax=lindata0.beta[0],
+        Betay=lindata0.beta[1],
+        Alphax=lindata0.alpha[0],
+        Alphay=lindata0.alpha[1],
+        Qpx=xsi[0],
+        Qpy=xsi[1],
+        A1=r1[0][0],
+        A2=r1[0][1],
+        A3=r1[1][1],
+        T1=-orbit,
+        T2=orbit,
+    )
     return nonlin_elem
+
+
+def feeddown_polynomba(
+    polb: numpy.ndarray or tuple = (),
+    pola: numpy.ndarray or tuple = (),
+    xoffset: float = 0,
+    yoffset: float = 0,
+    verbose: bool = False,
+    debug: bool = False,
+) -> dict[str, numpy.ndarray]:
+    """
+    Return the feeddown due to a transverse offset.
+
+    Parameters:
+        polb: numpy array. PolynomB.
+        pola: numpy array. PolynomA.
+        xoffset: float. Default zero. Horizontal offset in meters.
+        yoffset: float. Default zero. Vertical offset in meters.
+        verbose: prints additional info.
+        debug: prints information about the feeddown polynom construction.
+
+    Returns:
+        Dictionary with PolynomB and PolynomA feeddown components.
+
+    Raises:
+        ValueError: if none of the two polynoms is passed.
+
+    Note:
+        If only one polynom is passed, it is assumed to be PolynomB.
+    """
+    # check debug and verbose flags only once
+    debugprint = print if debug else lambda *a, **k: None
+    verboseprint = print if verbose else lambda *a, **k: None
+
+    # verify polynoms length
+    maxorda = len(pola)
+    maxordb = len(polb)
+    if maxorda == 0 and maxordb == 0:
+        raise ValueError("At least one polynom is needed")
+    maxord = max(maxorda, maxordb)
```

Comment:
Is this really useful? Why not keep the A/B calculation independent and allow for polynoms with different length
This way you can skip the whole zero padding part and simplify your code

Reply:
Sorry, I don't understand. The input polynom a and b already could be of different lengths.
On the other side, I think the output polynom a and b must be of the same length to avoid problems with the integrators. Therefore, I am returning something that could be assigned as Polynoms for an element. If I left the output polynoms with different length, it will be cumbersome to use them later.

Reply:
Ok up to you. Just so you know this "equalization" is already done in the element class when it is initialized see `ThinMultipole`

Reply:
@swhite2401 , thank you, I think I found the lines 
https://github.com/atcollab/at/blob/a409dd675bfa2075304049503c2c99da0726b3fb/pyat/at/lattice/elements.py#L781-L785.
I think I will keep the zero padding, and to avoid doing long calculations over a component that is equal to zero I have included an `if nthpolcomp != 0` in the last commit.